x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_response_schema.mock.ts

```diff
@@ -64,6 +64,7 @@ const getResponseBaseParams = (anchorDate: string = ANCHOR_DATE): SharedResponse
   timestamp_override: undefined,
   timestamp_override_fallback_disabled: undefined,
   namespace: undefined,
+  custom_highlighted_fields: undefined,
```

Comment:
I think it shouldn't be possible to set it to `undefined` in the response (`RuleResponse` and other rule type-specific types). It should always be an array (empty or non-empty), which would improve DX for API users (including us on the FE side).